sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp

```diff
@@ -51,6 +52,11 @@ Azure::Security::KeyVault::Keys::DeleteKeyOperation::PollInternal(
       m_value = _detail::DeletedKeySerializer::DeletedKeyDeserialize(m_value.Name(), *rawResponse);
     }
   }
+  else
```

Comment:
nit. Consider refactoring this method as a `top-returns` strategy instead of the `if-else`.  This would improve readability.

Something like:
```cpp
if (IsDone()) {
   return std::move(m_rawResponse);  
}

// The logic here is for isDone == false
try{
...
} catch (){
...
}
switch() {
...
}
return
```